src/barnetilsyn/BarnetilsynContext.tsx

```diff
@@ -95,18 +94,31 @@ const [BarnetilsynSøknadProvider, useBarnetilsynSøknad] = createUseContext(
     };
 
     const hentForrigeSøknadBarnetilsyn = async (): Promise<void> => {
-      return hentDataFraForrigeBarnetilsynSøknad().then(
-        (tidligereVersjon?: ForrigeSøknad) => {
-          if (tidligereVersjon) {
-            settSøknad((prevSøknad) => ({
-              ...prevSøknad,
-              ...tidligereVersjon,
-            }));
-          }
-        }
-      );
+      const forrigeSøknad = await hentDataFraForrigeBarnetilsynSøknad();
+      if (forrigeSøknad) {
+        settSøknad((prevSøknad) => ({
+          ...prevSøknad,
+          ...forrigeSøknad,
+          person: {
+            ...prevSøknad.person,
+            barn: [
+              ...forrigeSøknad.person.barn,
+              ...prevSøknad.person.barn.filter(
+                (barn) =>
+                  !forrigeSøknad.person.barn.some(
+                    (prevBarn) => prevBarn.ident.verdi === barn.ident.verdi
+                  )
+              ),
+            ],
+          },
+        }));
+      }
     };
 
+    useEffect(() => {
+      console.log('søknad i barnetilsynContext: ', søknad);
```

Comment:
Hvis dette er for lokal testing; husk å slett senere.
Evt. legg inn en `TODO: Slett denne`

Reply:
Det er planen. Se kommentaren over